models/Import/Factory/QtiResultXmlFactory.php

```diff
@@ -0,0 +1,244 @@
+use oat\taoResultServer\models\classes\ResultManagement;
+use oat\taoResultServer\models\classes\ResultServerService;
+use oat\taoResultServer\models\Import\Exception\ImportResultException;
+use oat\taoResultServer\models\Import\Input\ImportResultInput;
+use stdClass;
+use taoResultServer_models_classes_ReadableResultStorage;
+use taoResultServer_models_classes_ResponseVariable;
+use taoResultServer_models_classes_Variable;
+
+class QtiResultXmlFactory
+{
+    private Ontology $ontology;
+    private ResultServerService $resultServerService;
+
+    public function __construct(Ontology $ontology, ResultServerService $resultServerService)
+    {
+        $this->ontology = $ontology;
+        $this->resultServerService = $resultServerService;
+    }
+
+    /**
+     * @throws ImportResultException
+     * @throws common_exception_Error
+     * @throws core_kernel_persistence_Exception
+     */
+    public function createByImportResult(ImportResultInput $input): string
+    {
+        $itemResults = [];
+        $timestamp = (new DateTime())->format(DATE_RFC3339_EXTENDED);
+        $deliveryExecutionId = $input->getDeliveryExecutionId();
+
+        $resultStorage = $this->resultServerService->getResultStorage();
+
+        if (!$resultStorage instanceof ResultManagement) {
+            throw new ImportResultException(
+                sprintf(
+                    'ResultsStorage must implement %s. Instance of %s provided',
+                    ResultManagement::class,
+                    get_class($resultStorage)
+                )
+            );
+        }
+
+        $outcomeVariables = $resultStorage->getDeliveryVariables($deliveryExecutionId);
+
+        /** @var taoResultServer_models_classes_ResponseVariable $scoreTotalVariable */
+        $scoreTotalVariable = null;
+        $scoreTotal = null;
+        $scoreTotalMax = null;
+        $updatedScoreTotal = null;
+
+        foreach ($outcomeVariables as $id => $outcomeVariable) {
+            if (!is_array($outcomeVariable)) {
+                continue;
+            }
+
+            /** @var stdClass $variable */
+            $variable = current($outcomeVariable);
+
+            if (!is_object($variable) || !property_exists($variable, 'variable')) {
+                continue;
+            }
+
+            /** @var taoResultServer_models_classes_Variable $variable */
+            $variable = $variable->variable;
+
+            if (!$variable instanceof taoResultServer_models_classes_Variable) {
+                continue;
+            }
+
+            if ($variable->getIdentifier() === 'SCORE_TOTAL') {
+                $scoreTotal = (float)$variable->getValue();
+
+                $updatedScoreTotal = $scoreTotal;
+                $scoreTotalVariable = $variable;
+            }
+
+            if ($variable->getIdentifier() === 'SCORE_TOTAL_MAX') {
+                $scoreTotalMax = (float)$variable->getValue();
+            }
+        }
+
+        if (is_null($scoreTotal)) {
```

Comment:
I prefer to use `=== null` instead, because this will work a bit faster, that function call 😅